### DISCUSSION TITLE
WASM Linker AOT Build Fails

### DISCUSSION THREAD

**Discussion by andrey-shavelev** [unoplatform/uno#9124] [Q&A] [answered]
I'm trying to build the Uno Wasm app on mac with AOT. It's an InterpreterAndAOT mode. The build always fails. I tried to troubleshoot the problem for a few days already, but with no success. 
The error is:

`
  [00:00:23.0243995] ILLink: error IL1040: Method 'Windows.UI.Core.CoreDispatcher.RunAsync(CoreDispatcherPriority,DispatchedHandler)' reference could not be resolved 
`

Windows.UI.Core.CoreDispatcher is declared in the Uno.dll
However, adding this to the csproj has no effect:

```
<ItemGroup>
   <MonoRuntimeMixedModeExcludedAssembly Include="Uno" />
</ItemGroup>
```

Thanks for helping.

**Comment by jeromelaban** [comment]
Thanks for the report! This error generally means that you're importing an Uno 3.x compatible library in an Uno 4.x application. You may need to update some of your nuget dependencies to later versions.

Which packages are you using?

**Comment by andrey-shavelev** [comment, marked as answer]
Thanks for the advice jeromelaban,
Here is the list of UNO-related packages. I hope it will help to identify the problem:

```
    <PackageReference Include="Uno.Core" Version="4.0.1" />
    <PackageReference Include="Uno.Microsoft.Toolkit" Version="7.1.11" />
    <PackageReference Include="Uno.Microsoft.Toolkit.Uwp" Version="7.1.11" />
    <PackageReference Include="Prism.DryIoc.Uno" Version="8.1.97" />
    <PackageReference Include="Uno.UI.RemoteControl" Version="4.1.9" />
    <PackageReference Include="Uno.WindowsStateTriggers" Version="1.1.1-uno.132" />
    <PackageReference Include="SkiaSharp.Views.Uno" Version="2.88.0-preview.187" />
    <PackageReference Include="Uno.UI.WebAssembly" Version="4.1.9" />
    <PackageReference Include="Uno.UI.Lottie" Version="4.1.9" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="3.3.1" />
    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="3.3.1" />
```

There was an implicit reference to Uno.Core version 3. But updating directly referencing v4 did not solve the issue.

**Reply by jeromelaban** [reply]
You're referencing Prism, which does not yet have a public version referencing Uno 4.x. See [here for details](https://github.com/PrismLibrary/Prism/discussions/2708) about this.

**Reply by andrey-shavelev** [reply]
The answer from Prism guys is not very supportive. Nevertheless, the problem is clear now. Thanks!

**Reply by jeromelaban** [reply]
They're maintaining this project on their free time. It's their way of supporting the project, in the same way we in Uno Platform are offering paid support to bump issues up in our backlog.

**Reply by dansiegel** [reply]
@andrey-shavelev we work on Prism in our free time. It is a free product licensed under MIT "AS-IS". We would love to make working on Prism our full time jobs and get releases out on a regular cadence. Unfortunately like so many Open Source projects, we find people want to demand that we do things to support their business activities, but are unwilling to offer support to the project in any fashion. 

If this is important to your organization I might suggest that you look at either an Enterprise Support contract or stepping up as a Corporate Sponsor of the Prism Library.

**Reply by andrey-shavelev** [reply]
@dansiegel, your project makes life better for many people in the software development community. Thank you guys for your input. We will discuss the possible solutions with the team.

**Reply by brianlagunas** [reply]
@andrey-shavelev the code in the master branch is updated to Uno 4.3.8. just build the source and you're unblocked.
